src/symtab/symbol.hpp

```diff
@@ -348,6 +348,11 @@ class Symbol {
     bool is_variable() const noexcept;
 
     std::string to_string() const;
+
+    bool is_writable() const noexcept {
+        return has_any_property(syminfo::NmodlType::range_var) ||
+               has_any_property(syminfo::NmodlType::write_ion_var);
+    }
```

Comment:
@alkino :  
* `ASSIGNED` variables (`assigned_definition`) are also behave as a range by default. 
* Same thing for state variables
* Technically, IIRC, read ion variable can be also can be written by user (I think there exist some mod files
You can see the comments in the code here: https://github.com/BlueBrain/nmodl/blob/abdbecf364ea65aa2347cd3c6363691f7c6ea022/src/codegen/codegen_helper_visitor.cpp#L373